receiver/opencensusreceiver/config.go

```diff
@@ -72,13 +77,14 @@ type keepaliveEnforcementPolicy struct {
 }
 
 func (rOpts *Config) buildOptions() (opts []Option, err error) {
-	tlsCredsOption, hasTLSCreds, err := ToOpenCensusReceiverServerOption(rOpts.TLSCredentials)
-	if err != nil {
-		return opts, fmt.Errorf("error initializing OpenCensus receiver %q TLS Credentials: %v", rOpts.NameVal, err)
-	}
-	if hasTLSCreds {
-		opts = append(opts, tlsCredsOption)
+	if rOpts.TLSCredentials != nil {
+		tlsCredsOptions, er := rOpts.TLSCredentials.LoadgRPCTLSServerCredentials()
+		if er != nil {
+			return nil, fmt.Errorf("error initializing OpenCensus receiver %q TLS Credentials: %v", rOpts.NameVal, er)
+		}
+		opts = append(opts, WithGRPCServerOptions(tlsCredsOptions))
```

Comment:
Might be better to use all named return values or none at all to avoid having the `er` which might be easy to mix up.

```suggestion
	if rOpts.TLSCredentials != nil {
	         var tlsCredOptions someType
		tlsCredsOptions, err = rOpts.TLSCredentials.LoadgRPCTLSServerCredentials()
		if err != nil {
			return
		}
		opts = append(opts, WithGRPCServerOptions(tlsCredsOptions))
```

Then at the bottom can also just do a plain `return` to use the named arguments.

Reply:
Going to go to no named return values since when using `err` there it complains ```shadow: declaration of "err" shadows declaration at line 79 (govet)```

Reply:
I think it should only shadow if you're assigning err with `:=`, but not using named is fine as well.

Reply:
yeah but then the vars for tlsCredOptions looks a bit weird having it declared above when there is no error and it is the only case where an error can be returned.